src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -0,0 +1,160 @@
+  public String accessToken;
+
+  AribaConnector(AribaConnectorConfig config) {
+
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    try {
+      config.validateToken(collector);
+    } catch (AribaException | IOException e) {
+      LOG.error("Credentials are incorrect", e);
+    }
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    JsonArray jsonArray = listTemplates();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      JsonObject resultJsonObject = gson.fromJson(jsonArray.get(i), JsonObject.class);
+      String name = resultJsonObject.get("viewTemplateName").getAsString();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TEMPLATE).
+        canBrowse(false).canSample(true));
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
+    AribaServices aribaServices = new AribaServices(config);
+    try {
+      accessToken = aribaServices.getAccessToken();
+    } catch (AribaException e) {
+      LOG.error("Error in generating token", e);
+    }
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(io.cdap.plugin.common.ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String template = connectorSpecRequest.getPath();
+    if (template != null) {
+      properties.put(AribaPluginConfig.TEMPLATE_NAME, template);
+    }
+    try {
+      Schema schema = aribaServices.buildOutputSchema(accessToken, template, "prod");
+      specBuilder.setSchema(schema);
+    } catch (AribaException | InterruptedException e) {
+      LOG.error("Error in generating schema", e);
+    }
+    return specBuilder.addRelatedPlugin(new PluginSpec(AribaBatchSource.NAME, BatchSource.PLUGIN_TYPE,
+                                                       properties)).build();
+  }
+
+  private JsonArray listTemplates() {
+    AribaServices aribaServices = new AribaServices(config);
+    try {
+      accessToken = aribaServices.getAccessToken();
+    } catch (AribaException | IOException e) {
+      LOG.error("unable to generate access token", e);
+    }
+    URL viewTemplatesURL = HttpUrl.parse(config.getBaseURL()).
+      newBuilder()
+      .addPathSegments(METADATA_PATH)
+      .addPathSegments("prod/viewTemplates")
```

Comment:
take this value(prod/sandbox) as input and make a variable like "%s/viewTemplates" and use String.format

Reply:
Done